src/CitizenInitiative/CitizenInitiativeControllerSubscriber.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace AppBundle\CitizenInitiative;
+
+use AppBundle\Controller\EnMarche\CitizenInitiativeController;
+use AppBundle\Controller\EnMarche\CitizenInitiativeManagerContoller;
+use AppBundle\Entity\Adherent;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class CitizenInitiativeControllerSubscriber implements EventSubscriberInterface
+{
+    protected $router;
+    protected $tokenStorage;
+
+    public function __construct(Router $router, TokenStorageInterface $tokenStorage)
```

Comment:
Instead of typehinting with `Router` (which is a concrete complex class), you should just typehint against `UrlGeneratorInterface` as your code only needs to know about the `generate()` method.